floodlight/io/datasets.py

```diff
@@ -799,3 +801,278 @@ def _download_matches_info(self) -> None:
+
+                self._IDSSE_FILE_NAME_INFO = f"DFL_02_01_matchinformation_{competition}_DFL-MAT-{file_id}.{self._IDSSE_FILE_EXT}"
+                self._IDSSE_FILE_NAME_EVENT = f"DFL_03_02_events_raw_{competition}_DFL-MAT-{file_id}.{self._IDSSE_FILE_EXT}"
+                self._IDSSE_FILE_NAME_POSITION = f"DFL_04_03_positions_raw_observed_{competition}_DFL-MAT-{file_id}.{self._IDSSE_FILE_EXT}"
+
+                if not os.path.isfile(f"{self._data_dir}/{self._IDSSE_FILE_NAME_INFO}"):
+                    self._download_and_write(
+                        self._IDSSE_FILE_NAME_INFO, self._IDSSE_HOST_URL_INFO
+                    )
+                if not os.path.isfile(
+                    f"{self._data_dir}/{self._IDSSE_FILE_NAME_EVENT}"
+                ):
+                    self._download_and_write(
+                        self._IDSSE_FILE_NAME_EVENT, self._IDSSE_HOST_URL_EVENT
+                    )
+                if not os.path.isfile(
+                    f"{self._data_dir}/{self._IDSSE_FILE_NAME_POSITION}"
+                ):
+                    self._download_and_write(
+                        self._IDSSE_FILE_NAME_POSITION, self._IDSSE_HOST_URL_POSITION
+                    )
+
+    def get(
+        self,
+        match_id: str = "J03WMX",
+        teamsheet_home: Teamsheet = None,
+        teamsheet_away: Teamsheet = None,
+        events=True,
+        positions=True,
+    ) -> Tuple[
+        tuple[dict[str, dict[str, Events]], dict[str, Teamsheet], Pitch],
+        tuple[
+            dict[str, dict[str, XY]],
+            dict[str, Code],
+            dict[str, Code],
+            dict[str, Teamsheet],
+            Pitch,
+        ],
+    ]:
+
+        """Get event and position data from the IDSSE dataset.
+
+        Parameters
+        ----------
+        match_id : str, optional
+            Match name, check Notes section for valid arguments.
+            Defaults to the first match "J03WMX".
+        teamsheet_home: Teamsheet, optional
+            Teamsheet-object for the home team used to create link dictionaries of the
+                form `links[pID] = team`. If given as None (default), teamsheet is
+                extracted from the data.
+        teamsheet_away: Teamsheet, optional
+            Teamsheet-object for the home team used to create link dictionaries of the
+                form `links[pID] = team`. If given as None (default), teamsheet is
+                extracted from the data.
+        events: bool, optional
+            Specifies weather the event data should be returned. Default is True. If
+            false None will be returned instead of the event data objects.
+        positions: bool, optional
+            Specifies weather the position data should be returned. Default is True. If
+            false None will be returned instead of the event data objects. This will
+            improve performance considerably if only event data is required.
```

Comment:
cp error in the second row?

also, I understand the idea to use the word "positions" and "position data" here, but I think it should be at least clear that XY objcets are being returned.

Reply:
Thanks

Again, I will make changes to the returns and try to be more precise here!